app/controllers/todos_controller.rb

```diff
@@ -0,0 +1,46 @@
+class TodosController < ApplicationController
+  def index
+    #fetch all the todos
+    @todos = Todo.all.order([:deadline])
+  end
+
+  def show
+    #to show a todo
+    @todo = Todo.find(params[:id])
+  end
+
+  def new
+    #renders form new todos
+    @todo = Todo.new
+  end
+
+
+  def create
+    #create a new todo
+    @todo = Todo.create(todo_params)
+    redirect_to @todo
+  end
+
+  def edit
+    @todo = Todo.find(params[:id])
+  end
+
+  def update
+    #update an existing todo
+    @todo = Todo.find(params[:id])
+    @todo.update todo_params
+    redirect_to @todo
```

Comment:
For now, this will work. Try thinking of a scenario, where the updation part or even the creation part in general involves some complex functionalities. Where should we write this code?
Read more about MVC Architecture

Reply:
okay sure!